### PR TITLE
first version integrated for UI for pull request

### PR DESCRIPTION
**Pull request by marceines**
pull request for UI and first functionality

### CODE REVIEW

File: src/JCommander.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ */
+public class JCommander {
+
+    public static void main (String [] args)
+    {
+        try {
+            UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsLookAndFeel");
+        }
+        catch (Exception e) {
+            //JOptionPane.showMessageDialog(null,"Error al intentar cargar L&F");
```
Comment (edson-gonzales) [inline src/JCommander.java:16]: Commented code should be always removed

File: src/core/Folder.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class Folder extends Item {
+
+    public Folder (File file) {
+        super(file);
+    }
+
+    /**
+     * List of files and folders that are on a path
+     */
+    public List<String> listFilesAndFolders(String directoryName)
```
Comment (edson-gonzales) [inline src/core/Folder.java:19]: this is the an example of a how should be the documentation for java methods
/**
- Returns an Image object that can then be painted on the screen. 
- The url argument must specify an absolute {@link URL}. The name
- argument is a specifier that is relative to the url argument. 
- <p>
- This method always returns immediately, whether or not the 
- image exists. When this applet attempts to draw the image on
- the screen, the data will be loaded. The graphics primitives 
- that draw the image will incrementally paint on the screen. 
  *
- @param  url  an absolute URL giving the base location of the image
- @param  name the location of the image, relative to the url argument
- @return      the image at the specified URL
- @see         Image
  */
  public Image getImage(URL url, String name) {
        try {
            return getImage(new URL(url, name));
        } catch (MalformedURLException e) {
            return null;
        }
  }

Comment (edson-gonzales) [inline src/core/Folder.java:19]: The parameter directoryName, is only the name of the directory, is a relative path, or an absolute path?

File: src/core/Folder.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        File directory = new File(directoryName);
+
+        //get all the files from a directory
+        File[] fullFileList = directory.listFiles();
+
+        for (File file : fullFileList){
+            fileListed.add(file.getName());
+        }
+
+        return fileListed;
```
Comment (edson-gonzales) [inline src/core/Folder.java:31]: just an small correction, i think should be filesListed

File: src/core/Folder.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        super(file);
+    }
+
+    /**
+     * List of files and folders that are on a path
+     */
+    public List<String> listFilesAndFolders(String directoryName)
+    {
+        List<String> fileListed = new ArrayList<String>();
+        File directory = new File(directoryName);
```
Comment (edson-gonzales) [inline src/core/Folder.java:22]: What happen if the directory does not exists? usually when you are doing operation with files or directories it is a good practice to handle exceptions

File: src/core/Item.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    private double sizeKB;
+    private double sizeMB;
+    private double sizeGB;
+    private double sizeTB;
+    private Date lastModified;
+    private String filePermissions;
+
+    private Map<String, String> map;
+
+    public Item (java.io.File file) {
```
Comment (edson-gonzales) [inline src/core/Item.java:29]: Why are you adding the whole package name for java.io.File if you have imported that package?

File: src/core/Item.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+        if (file.isFile()) {
+            int i = file.getName().lastIndexOf('.');
```
Comment (edson-gonzales) [inline src/core/Item.java:32]: clean code rules, you should always have well-named variables

File: src/core/Item.java (hunk 5 above)
Comment (edson-gonzales) [inline src/core/Item.java:29]: This constructor has more than 20 lines of code

File: src/core/Item.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+            name = file.getName();
+            type = "";
+        }
+
+        lastModified = new Date(file.lastModified());
+        location = file.getPath();
+        sizeBytes = file.length();
+        sizeKB = (sizeBytes / 1024);
+        sizeMB = (sizeKB / 1024);
+        sizeGB = (sizeMB / 1024);
```
Comment (edson-gonzales) [inline src/core/Item.java:47]: all these file size logic looks like another method, remember methods should do one thing

File: src/core/Item.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        sizeTB = (sizeGB / 1024);
+
+        filePermissions = "";
+        if (file.canRead()) {
+            filePermissions += "r";
+        } else {
+            filePermissions += "-";
+        }
+        if (file.canWrite()) {
+            filePermissions += "w";
```
Comment (edson-gonzales) [inline src/core/Item.java:57]: same here this logic looks like another method

File: src/view/MenuBar.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    private JMenuItem filesChangeAttrib;
+    private JMenuItem filesCompareContent;
+    private JMenuItem filesProperties;
+    private JMenuItem filesQuit;
+
+    public MenuBar() {
+        initComponents();
+    }
+
+    private void initComponents() {
```
Comment (edson-gonzales) [inline src/view/MenuBar.java:29]: This method is bigger that 20 lines of code

File: src/view/MenuBar.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        configuration.setMnemonic(KeyEvent.VK_O);
+        help = new JMenu("Help");
+        help.setMnemonic(KeyEvent.VK_H);
+
+        filesChangeAttrib = new JMenuItem("Change Attributes", new ImageIcon(MenuBar.class.getResource("../image/iconAttributes.gif")));
+        filesChangeAttrib.addActionListener(this);
+        filesChangeAttrib.setMnemonic(KeyEvent.VK_A);
+        filesChangeAttrib.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_A, ActionEvent.ALT_MASK));
+        filesChangeAttrib.getAccessibleContext().setAccessibleDescription("Change attributes in selected folders/files");
+        filesCompareContent = new JMenuItem("Compare By Content", new ImageIcon(MenuBar.class.getResource("../image/iconCompare.png")));
```
Comment (edson-gonzales) [inline src/view/MenuBar.java:49]: try to use empty lines to separate logically the code, like after filesChangeAttrib add an empty line

File: src/view/PanelContainer.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Created by Marcela Barrionuevo on 16/05/2016.
+ */
+public class PanelContainer extends JPanel {
+
+    JTabbedPane jTabbedPane;
+    private String workingPath;
+
+    public PanelContainer(String workingPath) {
```
Comment (edson-gonzales) [inline src/view/PanelContainer.java:15]: documentation of the method

File: src/view/PanelContainer.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+        if (workingPath == null || workingPath.isEmpty()) {
+            workingPath = System.getProperty("user.dir");
+        }
+
+        jTabbedPane = new JTabbedPane();
+
+        JPanel jPaneA = new JPanel();
```
Comment (edson-gonzales) [inline src/view/PanelContainer.java:23]: why JPaneA? what it means?

File: src/view/TableSortedModel.java [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+
+        populateDataTable();
+    }
+
+    private void populateDataTable() {
+        data = new Object[items.size()][columnNames.length];
+
+        for (int index = 0; index < items.size(); index++) {
+            Item item = items.get(index);
+            data[index] = new Object[] {item.getName(), item.getType(), item.getSize(), item.getLastModified(), item.getFilePermissions()};
```
Comment (edson-gonzales) [inline src/view/TableSortedModel.java:35]: this is more than 100 characters you need to split up the line

File: src/view/TableSortedModel.java [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+        // Get all the folders/files
+        File fileSystem = new File(workingPath);
+
+        if (fileSystem.exists()) {
+            File[] fullFileList = fileSystem.listFiles();
+
+            for (File file : fullFileList){
+
+                Item item;
+                if (file.isDirectory()) {
```
Comment (edson-gonzales) [inline src/view/TableSortedModel.java:51]: more than 2 nested levels

File: src/view/ToolBar.java [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+        };
+
+        ImageIcon iconSave = new ImageIcon(MenuBar.class.getResource("../image/iconProperties.gif"));
+        Action actionSave = new AbstractAction("Properties", iconSave) {
+            public void actionPerformed(ActionEvent e) {
+                // TO DO
+            }
+        };
+
+        JButton btn1 = add(actionNew);
```
Comment (edson-gonzales) [inline src/view/ToolBar.java:38]: clean code, well named variables

**Comment by edson-gonzales** [comment]
Please review the observations made in the files, but in general review clean code rules and documentation of the methods according the example that i put in the code, after those corrections we will need another review

**Comment by marceines** [comment]
Thanks Edson, working on this...

**Pull request closed without merging**